src/main/scala/io/iohk/ethereum/domain/Transaction.scala

```diff
@@ -10,13 +10,16 @@ object Transaction {
   val GasLength = 32
   val ValueLength = 32
 
+  def apply(nonce: BigInt, gasPrice: BigInt, gasLimit: BigInt, receivingAddress: Address, value: BigInt, payload: ByteString): Transaction =
+    Transaction(nonce, gasPrice, gasLimit, Some(receivingAddress), value, payload)
+
 }
 
 case class Transaction(
   nonce: BigInt,
   gasPrice: BigInt,
   gasLimit: BigInt,
-  receivingAddress: Address,
+  receivingAddress: Option[Address],
```

Comment:
If we are going this way then we should remove `Address.empty` - it was supposed to handle the case of contract creation. Maybe an `Option` actually fits the domain better, but it would have been much easier to fix `Address.apply` to correctly produce `Address.empty`.